### PR TITLE
ELPP-2451 Added cache headers to search

### PR DESCRIPTION
**Pull request by stephenwf**

### CODE REVIEW

**Comment by stephenwf** [comment]
Added a web test case, I will do the same with Recommendations when it gets a web test framework installed.

**Comment by thewilkybarkid** [comment]
Could you also add one that checks that the `Cache-Control` header is correct?

**Comment by stephenwf** [comment]
added!

File: tests/src/Search/Web/CacheWebTest.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        $this->addDocumentsToElasticSearch([
+            $this->getArticleFixture(0),
+            $this->getArticleFixture(1),
+        ]);
+        $this->newClient();
+        $this->jsonRequest('GET', '/search');
+        $response = $this->getResponse();
+        $this->assertEquals(200, $response->getStatusCode());
+        $eTag = $response->headers->get('ETag');
+        $this->assertEquals('max-age=300, public', $response->headers->get('Cache-Control'));
```
Comment (thewilkybarkid) [inline tests/src/Search/Web/CacheWebTest.php]: Shouldn't this fail? `stale-while-revalidate` and `stale-if-error` should be there.

**Comment by thewilkybarkid** [comment]
Should check the `Accept` header too.

File: tests/src/Search/Web/CacheWebTest.php (hunk 1 above)
Comment (stephenwf) [inline tests/src/Search/Web/CacheWebTest.php]: That is my bad, added those to the wrong header!

**Pull request merged**